corehq/apps/userreports/sql.py

```diff
@@ -101,3 +104,33 @@ def _hash(domain, table_id):
 
     return 'config_report_{0}_{1}_{2}'.format(domain, table_id, _hash(domain, table_id))
 
+
+def get_expanded_columns(table_name, column_config):
+
+    session = Session()
+    connection = session.connection()
+    metadata = sqlalchemy.MetaData()
+    metadata.reflect(bind=connection)
+
+    column = metadata.tables[table_name].c[column_config.field]
+    query = sqlalchemy.select([column]).distinct()
+
+    result = connection.execute(query).fetchall()
```

Comment:
Using a connection should always be inside a try finally block that rolls back the transaction if an exception occurs. See https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/sqlreport.py#L254-254

Furthermore there is any chance that the code may be called outside of a django request then the session should be closed once it is not longer needed. Created a trello card for this too: https://trello.com/c/NdeoHnWq/74-better-handling-of-sqlalchemy-sessions